astropy/tests/helper.py

```diff
@@ -22,6 +24,7 @@
     "check_pickling_recovery",
     "pickle_protocol",
     "generic_recursive_equality_test",
+    "raises_with_notes",
```

Comment:
Not sure if this one should be in public API. When we bump Python and pytest minversions enough here, we will not need this one anymore. I want to be able to remove it without deprecation like our other compat stuff. Lots of downstream packages use stuff in test helper.

Thanks!

Reply:
oh, I naively assumed this module's content was for internal use only. So, my understanding is that I should
- add a trailing underscore to the function's name
- drop it from `__all__`
- anything else ?

Reply:
I wonder if we should move it to https://github.com/astropy/astropy/blob/main/astropy/utils/compat/misc.py and make pytest a local import. 🤔

Reply:
@mhvk , since you deal with compatibility stuff a lot lately, do you have a preference?

The thing with `utils` is that I am afraid the pytest import is going to come back and bite us.

Reply:
On the other hand, since this is only really used in two places out of the whole test suite, do we even need a function? One more set of if-else in the test code won't hurt.

Reply:
I'm not confident we can assume these will be the *only* two call sites, so I'd rather make it a function from the get go since it has just enough complexity that copy-paste-adapting the code is prone to error.

Reply:
If the immediate goal is to fix these 2 tests, I don't think we need this function. If we're going to add this function, it should be used consistently throughout the repo, not just these 2 tests. I think it is cleaner that way.

Reply:
What I meant was that, it should be used only to check exceptions that go through `astropy.utils.exceptions_add_note_to_exception`, which currently is just 2 tests, but there are already more than two call sites for this function and I have no reason to believe there couldn't be more. That said, doing a sweeping change to replace *all* calls to `pytest.raises` seems a bit excessive, doesn't it ?

Reply:
> seems a bit excessive

It indeed does.

Therefore here is my proposal: Avoid introducing new function and just do the if-else locally for the two affected tests. As this if-else would be removed eventually anyway, I am okay with it being a little messy. Thank you for your understanding!

Reply:
done